src/System.Private.CoreLib/shared/System/Text/Utf16Utility.cs

```diff
@@ -125,5 +125,39 @@ internal static bool UInt32ContainsAnyUppercaseAsciiChar(uint value)
 
             return (combinedIndicator & 0x00800080u) != 0;
         }
+
+        /// <summary>
+        /// Given two UInt32s that represent two ASCII UTF-16 characters each, returns true iff
+        /// the two inputs are equal using an ordinal case-insensitive comparison.
+        /// </summary>
+        /// <remarks>
+        /// This is a branchless implementation.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool UInt32OrdinalIgnoreCaseAscii(uint valueA, uint valueB)
+        {
+            // ASSUMPTION: Caller has validated that input values are ASCII.
+            Debug.Assert(AllCharsInUInt32AreAscii(valueA));
+            Debug.Assert(AllCharsInUInt32AreAscii(valueB));
+
+            // a mask of all bits which are different between A and B
+            uint differentBits = valueA ^ valueB;
+
+            // the 0x80 bit of each word of 'lowerIndicator' will be set iff the word has value < 'A'
+            uint lowerIndicator = valueA + 0x01000100u - 0x00410041u;
+
+            // the 0x80 bit of each word of 'upperIndicator' will be set iff (word | 0x20) has value > 'z'
+            uint upperIndicator = (valueA | 0x00200020u) + 0x00800080u - 0x007B007Bu;
+
+            // the 0x80 bit of each word of 'combinedIndicator' will be set iff the word is *not* [A-Za-z]
```

Comment:
I'm getting lost here. Why does this follow - how did you eliminate the characters that fall between `Z` and `a`, presumably using the `| 0x20` ?

Reply:
`A` = 0x41 = `0100 0001`
`Z` = 0x5A = `0101 1010`
`a` = 0x61 = `0110 0001`
`z` = 0x7A = `0111 1010`

0x20 = `0010 0000`

hmm

Reply:
Ah, I see, 0x20 makes the word "lowercase" obviously

Reply:
That might be worth noting after line 149

Reply:
I've updated the comment per your feedback and it'll be present in the next iteration.